Mage.Sets/src/mage/cards/s/ScionOfCalamity.java

```diff
@@ -0,0 +1,89 @@
+package mage.cards.s;
+
+import java.util.UUID;
+import mage.MageInt;
+import mage.abilities.TriggeredAbilityImpl;
+import mage.abilities.effects.common.DestroyTargetEffect;
+import mage.abilities.keyword.MyriadAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.SubType;
+import mage.constants.Zone;
+import mage.filter.FilterPermanent;
+import mage.filter.StaticFilters;
+import mage.filter.predicate.permanent.ControllerIdPredicate;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.players.Player;
+import mage.target.TargetPermanent;
+
+/**
+ * @author
+ */
+public final class ScionOfCalamity extends CardImpl {
+
+    public ScionOfCalamity(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{3}{G}{G}");
+
+        this.subtype.add(SubType.DINOSAUR);
+        this.power = new MageInt(5);
+        this.toughness = new MageInt(5);
+
+        // Myriad
+        this.addAbility(new MyriadAbility(true));
+
+        // Whenever Scion of Calamity deals combat damage to a player, destroy target artifact or enchantment that player controls.
+        this.addAbility(new ScionOfCalamityTriggeredAbility());
+    }
+
+    private ScionOfCalamity(final ScionOfCalamity card) {
+        super(card);
+    }
+
+    @Override
+    public ScionOfCalamity copy() {
+        return new ScionOfCalamity(this);
+    }
+}
+
+class ScionOfCalamityTriggeredAbility extends TriggeredAbilityImpl {
+
+    ScionOfCalamityTriggeredAbility() {
+        super(Zone.BATTLEFIELD, new DestroyTargetEffect(), false);
+    }
+
+    private ScionOfCalamityTriggeredAbility(final ScionOfCalamityTriggeredAbility ability) {
+        super(ability);
+    }
+
+    @Override
+    public ScionOfCalamityTriggeredAbility copy() {
+        return new ScionOfCalamityTriggeredAbility(this);
+    }
+
+    @Override
+    public boolean checkEventType(GameEvent event, Game game) {
+        return event.getType() == GameEvent.EventType.DAMAGED_PLAYER;
+    }
+
+    @Override
+    public boolean checkTrigger(GameEvent event, Game game) {
+        Player opponent = game.getPlayer(event.getPlayerId());
+        if (opponent != null && event.getSourceId().equals(this.sourceId)) {
```

Comment:
need to check is combat damage. (I guess Sunder Shaman is bugged the same way, but Aberrant and Trygon Predator get it right. Annoying that each has a different code style.)

Reply:
Fixed Sunder Shaman & Scion of Calamity to match Trygon Predator